File: test-suite-python/src/test/python/micronaut/docs/metrics/exception/OutOfStockExceptionHandler.py
from jakarta.inject import Singleton
from micronaut.context.annotation import Requires
from micronaut.http import HttpRequest, HttpResponse
from micronaut.http.server.exceptions import ExceptionHandler

from .OutOfStockException import OutOfStockException


@Requires(property="spec.name", value="OutOfStockExceptionTest")
@Singleton
class OutOfStockExceptionHandler(ExceptionHandler[OutOfStockException, HttpResponse]):

    def handle(self, request: HttpRequest, exception: OutOfStockException) -> HttpResponse:
        return exception.getResponse()

File: test-suite/src/test/java/io/micronaut/docs/metrics/exception/OutOfStockExceptionHandler.java
package io.micronaut.docs.metrics.exception;

import io.micronaut.context.annotation.Requires;
import io.micronaut.http.HttpRequest;
import io.micronaut.http.HttpResponse;
import io.micronaut.http.server.exceptions.ExceptionHandler;
import jakarta.inject.Singleton;

@Requires(property = "spec.name", value = "OutOfStockExceptionTest")
@Singleton
class OutOfStockExceptionHandler implements ExceptionHandler<OutOfStockException, HttpResponse<?>> {

    @Override
    public HttpResponse<?> handle(HttpRequest request, OutOfStockException exception) {
        return exception.getResponse();
    }
}
